src/web/components/CMP.tsx

```diff
@@ -1,35 +1,43 @@
 import React, { useState, useEffect } from 'react';
-import {
-    shouldShow,
-    setErrorHandler,
-} from '@guardian/consent-management-platform';
-import { ConsentManagementPlatform } from '@guardian/consent-management-platform/lib/ConsentManagementPlatform';
-
-type Props = {
-    cmpUi: boolean; // A switch to decide if we show CMP or not
-};
+import loadable from '@loadable/component';
+
+const ConsentManagementComponent = loadable.lib(() =>
+    import(
+        '@guardian/consent-management-platform/lib/ConsentManagementPlatform'
+    ),
+);
 
-export const CMP = ({ cmpUi }: Props) => {
+export const CMP = () => {
     const [show, setShow] = useState(false);
 
     useEffect(() => {
-        if (shouldShow()) {
-            setShow(true);
-
-            // setErrorHandler takes function to be called on errors in the CMP UI
-            setErrorHandler((errMsg: string): void => {
-                const err = new Error(errMsg);
-
-                window.guardian.modules.sentry.reportError(err, 'cmp');
-            });
-        }
+        import(
+            /* webpackChunkName: "consent-management-platform" */ '@guardian/consent-management-platform'
+        ).then(({ shouldShow, setErrorHandler }) => {
+            if (shouldShow()) {
+                setShow(true);
+
+                // setErrorHandler takes function to be called on errors in the CMP UI
+                setErrorHandler((errMsg: string): void => {
+                    const err = new Error(errMsg);
+
+                    window.guardian.modules.sentry.reportError(err, 'cmp');
+                });
+            }
+        });
     }, []);
 
     const onClose = () => setShow(false);
 
-    if (!show || !cmpUi) {
+    if (!show) {
         return null;
     }
 
-    return <ConsentManagementPlatform source="dcr" onClose={onClose} />;
+    return (
+        <ConsentManagementComponent>
+            {({ ConsentManagementPlatform }) => (
+                <ConsentManagementPlatform source="dcr" onClose={onClose} />
+            )}
+        </ConsentManagementComponent>
```

Comment:
Render props pattern?

Reply:
yeah this is how you can achieve lazy lib loading https://loadable-components.com/docs/library-splitting/